alicloud/monitoring_metric.go

```diff
@@ -155,6 +155,13 @@ func listCMMetricStatistics(ctx context.Context, d *plugin.QueryData, granularit
 	}
 
 	var results []map[string]interface{}
+	
+	// As some point of the time we are getting the error in response not in the error part.
+	// Response in stats variable: "%!v(PANIC=String method: runtime error: invalid memory address or nil pointer dereference)"
+	if stats.Datapoints == "" {
```

Comment:
@ParthaI Thinking about this a bit more, could we include this as part of the retry function again and if we get the intermittent error response above, retry (assuming the API usually behaves properly afterward)? I think returning `nil` is ok, but it then looks like there are no metric statistics, when in reality there may be.

A key point for the proposed solution above to work is that we need to be able to recognize that we're getting the intermittent error response from Alicloud and not an actual error/empty set of data points.

Reply:
@cbruno10 We have added the retry logic, if we are getting the error as an api response then it will retry it again and will get back the expected response, hopefully we will not face the subjected issue any more.